src/services/webhook-handler/webhook-sender.service.ts

```diff
@@ -0,0 +1,266 @@
+import { WebhookDeliveryStatus } from '@prisma/client';
+import { prisma } from '@/utils/db';
+import { logger } from '@/utils/logger';
+import { WebhookPayload } from '@/types/webhook-payloads';
+
+export interface WebhookDeliveryResult {
+  success: boolean;
+  responseCode?: number;
+  errorMessage?: string;
+  durationMs: number;
+}
+
+/**
+ * Service for sending individual webhook HTTP requests
+ */
+export class WebhookSenderService {
+  private static readonly REQUEST_TIMEOUT = 10000;
+  private static readonly USER_AGENT = 'Masumi-Webhook/1.0';
+
+  /**
+   * Send a webhook to a specific URL
+   */
+  async sendWebhook(
+    url: string,
+    authToken: string,
+    payload: WebhookPayload,
+  ): Promise<WebhookDeliveryResult> {
+    const startTime = Date.now();
+
+    try {
+      logger.info('Sending webhook', {
+        url,
+        event_type: payload.event_type,
+      });
+
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${authToken}`,
+          'X-Masumi-Event': payload.event_type,
+          'X-Masumi-Timestamp': payload.timestamp,
+          'User-Agent': WebhookSenderService.USER_AGENT,
+        },
+        body: JSON.stringify(payload),
+        signal: AbortSignal.timeout(WebhookSenderService.REQUEST_TIMEOUT),
+      });
+
+      const durationMs = Date.now() - startTime;
+
+      if (response.ok) {
+        logger.info('Webhook delivered successfully', {
+          url,
+          event_type: payload.event_type,
+          status_code: response.status,
+          duration_ms: durationMs,
+        });
+
+        return {
+          success: true,
+          responseCode: response.status,
+          durationMs,
+        };
+      } else {
+        const errorMessage = `HTTP ${response.status}: ${response.statusText}`;
+
+        logger.warn('Webhook delivery failed', {
+          url,
+          event_type: payload.event_type,
+          status_code: response.status,
+          error: errorMessage,
+          duration_ms: durationMs,
+        });
+
+        return {
+          success: false,
+          responseCode: response.status,
+          errorMessage,
+          durationMs,
+        };
+      }
+    } catch (error) {
+      const durationMs = Date.now() - startTime;
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+
+      logger.error('Webhook delivery error', {
```

Comment:
check if it is an timeout or htp error and do not log it as an error in that case